### PR TITLE
add support for graphql-yoga v3 executor

### PR DESCRIPTION
**Pull request by khanayan123**
### What does this PR do?
adds support for graphql-yoga v3 executor

### Motivation
customer request: https://github.com/DataDog/dd-trace-js/issues/2884

### Additional Notes
it was a bit challenging to figure out which v3 version of graphql-yoga added @graphql-tools/executor for execution. Upon testing it seems like it was added with 3.6.0^ minor release

### CODE REVIEW

**Comment by khanayan123** [comment]
the presence of graphql@^15.2.0 in versions causes the apollo core test to fail

**Review by Qard — APPROVED** [review]

File: packages/datadog-instrumentations/src/helpers/hooks.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ module.exports = {
   'node:fs': () => require('../fs'),
   'generic-pool': () => require('../generic-pool'),
   'graphql': () => require('../graphql'),
+  '@graphql-tools/executor': () => require('../graphql'),
```
Comment (rochdev) [inline packages/datadog-instrumentations/src/helpers/hooks.js:45]: Let's organize this alphabetically to make it easier to read.

File: packages/datadog-instrumentations/src/graphql.js [hunk 2]
```diff
@@ -350,6 +350,11 @@ function finishResolvers ({ fields }) {
   })
 }
 
+addHook({ name: '@graphql-tools/executor', file: 'cjs/execution/execute.js', versions: ['>=0.0.14'] }, execute => {
```
Comment (rochdev) [inline packages/datadog-instrumentations/src/graphql.js:353]: Does this mean that ESM is not supported?

File: packages/datadog-plugin-graphql/test/index.spec.js [hunk 3]
```diff
@@ -161,146 +164,225 @@ describe('Plugin', () => {
           return a.start.toString() >= b.start.toString() ? 1 : -1
         })
       })
-
-      describe('without configuration', () => {
-        before(() => {
-          return agent.load('graphql')
-            .then(() => {
+      // graphql-yoga@^3.6.0 requires graphql@>=15.2.0
+      if (semver.satisfies(version, '>=15.2.0')) {
```
Comment (rochdev) [inline packages/datadog-plugin-graphql/test/index.spec.js:168]: Any way to add this without that many changes to the file?

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/datadog-instrumentations/src/graphql.js (hunk 2 above)
Comment (khanayan123) [reply packages/datadog-instrumentations/src/graphql.js:353]: Haven't tested it with ESM yet. But this is more of a feature request to support graphql-yoga v3 https://github.com/DataDog/dd-trace-js/issues/2884 which uses @graphlql-tools/executor as it's executor rather the the graphql executor

File: packages/datadog-plugin-graphql/test/index.spec.js [hunk 4]
```diff
@@ -1535,91 +1614,92 @@ describe('Plugin', () => {
...
-            schema,
-            query: 'query MyQuery { hello }',
-            operationName: 'MyQuery'
-          }
-
-          runQuery(params)
-            .catch(done)
-        })
-      })
+      // TODO: this is an outdated test, comeback to it to officially add support for
```
Comment (tlhunter) [inline packages/datadog-plugin-graphql/test/index.spec.js:1617]: Can you elaborate on this test removal more? At first glance it looks like this PR breaks support for `apollo-server-core`.

Comment (rochdev) [reply packages/datadog-plugin-graphql/test/index.spec.js:1617]: Let's also call `.skip` on the test instead of commenting it out.

Comment (khanayan123) [reply packages/datadog-plugin-graphql/test/index.spec.js]: @tlhunter 

The precense of graphql@^15.2.0 in the /versions folder causes graphql-tools@3.1.1
to break in the before() hook. This test tests a library that had it's last release occur 5 years ago &
updating the test would require using newer version of apollo-core which have a completely different syntax
and produces traces that are different then what is expected by this test

**Review by tlhunter — APPROVED** [review]

**Review by Qard — APPROVED** [review]

**Review by rochdev — APPROVED** [review]

**Pull request merged**